test/helpers/kind/setup.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
```

Comment:
Is there any room for non-kind based e2e tests? it would be good if we didn't bind ourselves to kind in the package name when it's an implementation detail. The one way I could see a package name of `kind` being good is if we have multiple ways to spin up a management cluster and kind is one of those ways.

Reply:
I believe there may be, but I would expect them to implement their own helper.